plugins/optimization/src/main/java/org/qbicc/plugin/opt/ea/EscapeAnalysisState.java

```diff
@@ -0,0 +1,63 @@
+package org.qbicc.plugin.opt.ea;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.qbicc.context.AttachmentKey;
+import org.qbicc.context.CompilationContext;
+import org.qbicc.graph.Call;
+import org.qbicc.graph.New;
+import org.qbicc.type.definition.element.ExecutableElement;
+
+final class EscapeAnalysisState {
+    private static final AttachmentKey<EscapeAnalysisState> KEY = new AttachmentKey<>();
```

Comment:
I think we really want this to be a `PhaseAttachmentKey` so that the state is updated on every phase.  As the program gets smaller, the environment around values which previously escaped might be transformed so that the values no longer escape.  In general we should be ensuring that *every analysis* is repeatable between phases.

This would allow us, for example, to use the escape information from the *previous* phase within peephole optimizations in an unrelated BBB.

Reply:
I've switched that now. Since `EscapeAnalysisOptimizeVisitor` runs on `LOWER`, it now asks for the previews phase escape state and uses that.